### PR TITLE
LPS-150933, LPS-162107, LPS-170861 | master

### PR DESCRIPTION
**Pull request by gaborlovasdotcom**
https://issues.liferay.com/browse/LPS-150933 - Feature flag
https://issues.liferay.com/browse/LPS-162107 - Functional test automation
https://issues.liferay.com/browse/LPS-170861 - Functional test automation

cc: @stian-sigvartsen, @JoyceWang08211

### CODE REVIEW

**Comment by gaborlovasdotcom** [comment]
ci:stop

**Comment by gaborlovasdotcom** [comment]
ci:test:sf

**Comment by gaborlovasdotcom** [comment]
ci:test:security

**Comment by gaborlovasdotcom** [comment]
ci:test:relevant

File: build-test-saml.xml [hunk 1]
```diff
@@ -3,9 +3,9 @@
 <project basedir="." name="portal-test-saml" xmlns:antelope="antlib:ise.antelope.tasks">
 	<import file="build-test.xml" />
 
-	<target name="deploy-keystore-file">
+	<target name="deploy-SAML-keystore-file">
 		<copy
-			file="${project.dir}/portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructureee/security/dependencies/keystore.jks"
+			file="${project.dir}/portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructureee/security/dependencies/SAMLKeystore.jks"
```
Comment (JoyceWang08211) [inline build-test-saml.xml:8]: The portal cannot recognize the jks certificate if it is changed to a name other than keystore.jks
See failures:
https://testray.liferay.com/home/-/testray/case_results/2242821270
https://testray.liferay.com/home/-/testray/case_results/2242821201
https://testray.liferay.com/home/-/testray/case_results/2242821221
https://testray.liferay.com/home/-/testray/case_results/2242821429

**Review by JoyceWang08211 — CHANGES_REQUESTED** [review]
I think we need to rename the jks certificate to keystore.jks. Other parts look good to me.

File: build-test-saml.xml (hunk 1 above)
Comment (gaborlovasdotcom) [reply build-test-saml.xml:8]: @stian-sigvartsen Is this filename is hardcoded?

Comment (gaborlovasdotcom) [reply build-test-saml.xml]: Solved, I ignored this change.

**Comment by gaborlovasdotcom** [comment]
ci:sf

**Comment by gaborlovasdotcom** [comment]
ci:forward

**Comment by gaborlovasdotcom** [comment]
ci:stop:relevant

**Comment by gaborlovasdotcom** [comment]
ci:forward:force

**Comment by gaborlovasdotcom** [comment]
ci:stop

**Comment by gaborlovasdotcom** [comment]
ci:forward

**Comment by gaborlovasdotcom** [comment]
ci:forward:force

**Comment by JoyceWang08211** [comment]
ci:forward:force

**Comment by JoyceWang08211** [comment]
ci:test:relevant

**Comment by gaborlovasdotcom** [comment]
ci:forward:force

**Comment by gaborlovasdotcom** [comment]
Forwarded directly in: https://github.com/brianchandotcom/liferay-portal/pull/132365

**Pull request closed without merging**